File: Program7.java
/*
	Write a program to find factorial of given number.
*/

import java.util.Scanner;

class Numbers
{

	private int iNum;
	
	public Numbers(int iValue)
	{
		iNum = iValue;
	}
	
	public int factorial()
	{
		int iCnt = 0,iFact = 1;
		
		if(iNum < 0)
		{
			iNum = -iNum;
		}
		
		for(iCnt = 1; iCnt <= iNum; iCnt++)
		{
			iFact = iFact * iCnt;
		}
		return iFact;
	
	}
	
}

class Program7
{

	public static void main(java.lang.String[] Arg)
	{	
		int iNo = 0;
	
		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter Number : ");
		iNo = sObj.nextInt();
		
		Numbers nObj = new Numbers(iNo);
		
		System.out.println("Factorial of "+iNo+" is : "+nObj.factorial());
				
		sObj.close();		
		
	}
}

File: Program7.c
/*
	Write a program to find factorial of given number.
*/

#include<stdio.h>

int factorial(int iNum)
{

	register int iCnt = 0;
	auto int iFact = 1;
	
	if(iNum < 0)
	{
		iFact = -iFact;
	}
	
	for(iCnt = 1; iCnt <= iNum; iCnt++)
	{
		iFact = iFact * iCnt;
	}
	
	return iFact;

}

int main(void)
{
	
	int factorial(int);
	
	auto int iNo = 0;
	
	printf("Enter number : ");
	scanf("%d",&iNo);
	
	printf("Factorial of %d is : %d\n",iNo,factorial(iNo));

	return 0;

}
